src/foam/nanos/auth/PasswordExpiryAuthService.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'PasswordExpiryAuthService',
+  extends: 'foam.nanos.auth.ProxyAuthService',
+
+  documentation: 'Checks password expiry',
+
+  imports: [
+    'localUserDAO'
+  ],
+
+  javaImports: [
+    'javax.naming.AuthenticationException',
+    'java.util.Calendar',
+    'java.util.TimeZone'
+  ],
+
+  methods: [
+    {
+      name: 'login',
+      javaCode:
+`User user = getDelegate().login(x, userId, password);
+ if ( isPasswordExpired(user) ) {
+   throw new AuthenticationException("Password expired");
+ }
+ return user;`
+    },
+    {
+      name: 'loginByEmail',
+      javaCode:
+`User user = getDelegate().loginByEmail(x, email, password);
+if ( isPasswordExpired(user) ) {
+  throw new AuthenticationException("Password expired");
+}
+return user;`
+    },
+    {
+      name: 'isPasswordExpired',
+      documentation: 'Checks if password is expired. True if expired, false if not expired',
+      javaReturns: 'boolean',
+      args: [
+        {
+          name: 'user',
+          javaType: 'foam.nanos.auth.User'
+        }
+      ],
+      javaCode:
+`if ( user == null ) return true;
+if ( user.getPasswordExpiry() == null ) return false;
+
+// if we are after the expiry date then prevent login
```

Comment:
Does this do the same thing:?
return user.getPasswordExpiry().getTime() < System.currentTimeMillis();

Reply:
Or what about user.getPasswordExpiry().compareTo(new Date()) ?

Reply:
Second version creates a new object whereas the first version doesn't.